samples/LocalizationSample/Startup.cs

```diff
@@ -10,18 +10,38 @@
 using Microsoft.AspNetCore.Localization;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Options;
+using MyLibrary;
+using OtherNamespace;
 
 namespace LocalizationSample
 {
     public class Startup
     {
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddLocalization(options => options.ResourcesPath = "My/Resources");
+            services.AddOptions();
+            services.TryAddSingleton(typeof(IStringLocalizerFactory), (serviceProvider) =>
```

Comment:
Could we configure `ClassLibraryStringLocalizerFactory` via options? Seems gross to do `GetRequiredService`